src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -42,27 +52,26 @@ export class LogStreamRegistry {
         parseCloudWatchLogsUri(uri)
         if (!this.hasLog(uri)) {
             this.setLogData(uri, initialStreamData)
-            this.registerLogHandlers(uri)
             await this.updateLog(uri, 'tail')
         }
     }
 
-    public getOnCloseFuncForUri(uri: vscode.Uri): (document: vscode.TextDocument) => void {
-        return (document: vscode.TextDocument) => {
-            if (document.uri.toString() === uri.toString()) {
-                this.clearStreamIdMap(uri)
-            }
+    public cleanUpDocument(document: vscode.TextDocument): void {
```

Comment:
"dispose" is the common term, let's use that for discoverability.

```suggestion
    public disposeDocument(document: vscode.TextDocument): void {
```